res/css/photon/checkbox.css

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+.photon-checkbox {
+  position: relative;
+  padding: 0;
+  border: 1px solid var(--grey-90-a30);
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  appearance: none;
+  background-color: var(--grey-90-a10);
+  border-radius: 2px;
+}
+
+/* Use this class with the photon-checkbox for the canonical photon styling. */
+.photon-checkbox-default {
+  width: 16px;
+  height: 16px;
+  margin: 0 4px;
+}
+
+/* Use this class with the photon-checkbox for the custom micro variant. */
+.photon-checkbox-micro {
+  width: 13px;
+  height: 13px;
+  margin: 0 3px;
+}
+
+.photon-checkbox:hover {
+  background-color: var(--grey-90-a20);
+}
+
+.photon-checkbox:checked {
+  border: none;
+  background-color: var(--blue-60);
+}
+
+.photon-checkbox:hover:active {
+  background-color: var(--grey-90-a30);
+}
+
+.photon-checkbox-default:checked {
+  background-image: url(../../img/svg/checkbox-check-16.svg);
+}
+
+.photon-checkbox-micro:checked {
+  background-image: url(../../img/svg/checkbox-check-13.svg);
+}
```

Comment:
I'd prefer to keep the custom svgs, rather than work around styling sizes using CSS, as it makes it easy to ensure pixel snapiness, as these shapes are designed on a grid, and avoid potentially blurry shapes.

Reply:
For me, using the CSS background-size is using SVG like they're designed for, it's not working around.

Moreover:
* for the 13px version of the radio button, you used fractions of pixels.
* for the 2 versions of the checkbox images we see fractions in the path as well.

I think that's why the 13px versions you did aren't more snappy than the scaled down versions of the 16px. Also these shapes are more round than squared, I think this is what makes it possible too.
I agree that for more squared and aligned icons there could be indeed a difference, but I don't think that's the case here. I didn't try on a high-dpi screen, but on my normal screen there was no perceivable difference.